atom/browser/api/atom_api_web_contents.cc

```diff
@@ -1489,13 +1489,21 @@ int WebContents::GetFrameRate() const {
 }
 
 void WebContents::Invalidate() {
-  if (!IsOffScreen())
-    return;
-
-  auto* osr_rwhv = static_cast<OffScreenRenderWidgetHostView*>(
+  if (IsOffScreen()) {
+    auto* osr_rwhv = static_cast<OffScreenRenderWidgetHostView*>(
       web_contents()->GetRenderWidgetHostView());
-  if (osr_rwhv)
-    osr_rwhv->Invalidate();
+    if (osr_rwhv)
+      osr_rwhv->Invalidate();
+  } else {
+    const auto ownerWindow = owner_window();
+    const auto nativeWindow = ownerWindow ? ownerWindow->GetNativeWindow() :
+                                            nullptr;
+    if (nativeWindow) {
+      const gfx::Rect& bounds = nativeWindow->bounds();
+      nativeWindow->SchedulePaintInRect(
```

Comment:
Could this just use `SchedulePaint()` which paints the whole view, https://cs.chromium.org/chromium/src/ui/views/view.cc?l=831&rcl=54520ce5e56cd898668f0f86172cdeb9914a9b71